doc/1.4/language.md

```diff
@@ -1879,6 +1882,97 @@ be supported.
+detached from the hook, and then executed &mdash; receiving the message as data.
+Computations suspended on a hook are executed in order of least recently
+attached.
+
+Currently, the only computations that can be suspended and attached to hooks are
+single method calls, which is done through the use of the [`after`
+statement](#after-statements). This will later be expanded upon: hooks will play
+a central role in the future introduction of *coroutines*, as hooks will serve
+as the primitive mechanism through which coroutines suspend themselves and
+become resumed.
+
+Every hook has an associated list of *message component types*, specified during
+declaration through the <tt>(<em>msgtype1</em>, ... <em>msgtypeN</em>)</tt>
+syntax. This specifies what form of data is sent and received via the hook. Any
+number of message component types can be given, including zero, in which case a
+message sent via the hook has no associated data. If exactly one message
+component type is specified the parentheses can be omitted.
+
+As shown in the second hook declaration form given above, it is possible to
+declare hook arrays. Unlike for [object declarations](#object-declarations), the
+dimensions of a hook array do not have associated index parameters, and so each
+dimension is specified only through its size. Every member of a hook array is
+distinct; sending a message to one hook of the array will not send a message to
+the rest.
+
+Example declarations:
+```
+// Hook with no associated message component types
+hook () h1;
+// Hook with a single message component type
+hook int h2;
+// Two-dimensional hook array with two message component types
+hook (int *, bool) h3[5][3];
+```
+
+Beyond suspending computations on it, a hook <tt><em>h</em></tt> has two
+associated operations:
+
+* <pre><em>h</em>.send_now(<em>msg1</em>, ... <em>msgN</em>)</pre>
+  Sends a message through the hook, with message components
+  <tt><em>msg1</em></tt> through <tt><em>msgN</em></tt>. The number of message
+  components must match the number of message component types of the hook,
+  and each message component must be compatible with the corresponding message
+  component type of the hook.
+
+  `send_now` is *synchronous*: every computation suspended on the hook will
+  execute before `send_now` completes.
+  `send_now` returns the number of suspended computations that were successfully
+  resumed from the message being sent. Currently, with the only possible
+  suspended computations being method calls suspended via `after`, every
+  suspended computation is guaranteed to successfully be resumed unless
+  cancelled by a preceding computation resumed by the `send_now`. This will not
+  remain true in the future: coroutines are planned to be able to reject a
+  message and reattach themselves to the hook.
+
+  <div class="note">
+  <b>Note:</b> Hooks are planned to be extended with an asynchronous variant of
```

Comment:
note: can also be written
 ```
**Note:**
```

Reply:
I _tried_ that, but it went wrong when viewed in the Simics documentation framework for whatever reason.

Reply:
Ah right, because you are inside a HTML element, and in there everything is html with no markdown extensions.